urld1e4af3f63_GMIyer_myastro_tgz-pJ8-loader_MatchMakerMainJ8/src/myastro/service/matchmaker/impl/RashiAthipatiKootaService.java

```diff
@@ -0,0 +1,196 @@
+package myastro.service.matchmaker.impl;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import myastro.constants.Graha;
+import myastro.constants.Rashi;
+import myastro.constants.matchmaker.MatchResult;
+import myastro.service.matchmaker.IRashiKootaService;
+
+public class RashiAthipatiKootaService implements IRashiKootaService {
+
+	private static final RashiAthipatiKootaService RASHI_ATHIPATI_KOOTA_SERVICE = new RashiAthipatiKootaService();
+
+	public static IRashiKootaService getCheckRashiAthipatiKootaService() {
+		return RASHI_ATHIPATI_KOOTA_SERVICE;
+	}
+
+	private final Map<Graha, HashMap<Graha, GrahaRelation>> rashiAthipatiDetailsCache;
+
+	private RashiAthipatiKootaService() {
+		rashiAthipatiDetailsCache = getRashiAthipatiDetails();
+	}
+
+	@Override
+	public MatchResult checkKoota(Rashi boyRashi, Rashi girlRashi) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 36-43 into a separate method.